### PR TITLE
[#67] Header navigation

### PR DESCRIPTION
**Pull request by nathan-schmidt-viget**
# Summary

This styles the header navigation. Removed the WP default a11y as it was not fully 100% a11y and built my own with AlpineJS. Because we can't access some of the navigation HTML I had to inject the Alpine with JS onto the DOM. 

The dark version will just change if we add `dark` to the parent div in the parts. 

The styles are pretty basic allowing us to customize per client. 

## Issues

* #67 

## Testing Instructions

1. Add navigation items in the admin. 
2. Test out that navigation on desktop and mobile. 
3. Test with keyboard and screen reader to make sure everything is a11y. 

## Screenshots
| | |
|--------|--------|
| Screenshot 2024-05-21 at 8 13 18 AM | Screenshot 2024-05-21 at 8 13 04 AM |
| Screenshot 2024-05-21 at 8 14 34 AM | Screenshot 2024-05-21 at 8 15 02 AM |

https://github.com/vigetlabs/wordpress-site-starter/assets/91974372/bd44aba4-419a-4f7c-94b5-f510e879f563

### CODE REVIEW

**Review by ten1seven — APPROVED** [review]

File: wp-content/themes/wp-starter/blocks/navigation-container/render.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			class="hidden"
+		<?php endif; ?>
+		aria-rel="mobile-toggle"
+	>
+		<svg
+			<?php if ( ! is_admin() ) : ?>
+				x-show="!menuIsOpen"
+			<?php endif; ?>
+			class="m-2" xmlns="http://www.w3.org/2000/svg" width="27" height="18" viewBox="0 0 27 18" fill="currentColor" class="menu-open">
+			<title>Open Menu</title>
```
Comment (bd-viget) [inline wp-content/themes/wp-starter/blocks/navigation-container/render.php:34]:
```suggestion
			<title><?php esc_html_e( 'Open Menu', 'wp-starter' ); ?></title>
```
Make this translatable

File: wp-content/themes/wp-starter/blocks/navigation-container/render.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			<path d="M26.8059 1H0.805908" stroke="currentColor" stroke-width="1.5"></path>
+			<path d="M26.8059 17H0.805908" stroke="currentColor" stroke-width="1.5"></path>
+			<path d="M26.8059 9H0.805908" stroke="currentColor" stroke-width="1.5"></path>
+		</svg>
+		<svg
+			<?php if ( ! is_admin() ) : ?>
+				x-show="menuIsOpen"
+			<?php endif; ?>
+			class="m-2" xmlns="http://www.w3.org/2000/svg" width="21" height="20" fill="currentColor" class="menu-close">
+			<title>Close Menu</title>
```
Comment (bd-viget) [inline wp-content/themes/wp-starter/blocks/navigation-container/render.php:44]:
```suggestion
			<title><?php esc_html_e( 'Close Menu', 'wp-starter' ); ?></title>
```
Support translations

File: wp-content/themes/wp-starter/blocks/navigation-container/render.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	>
+		<div
+			<?php if ( ! is_admin() ) : ?>
+				x-data="dropdown"
+				x-on:keydown.escape.prevent.stop="close($refs.button)"
+				x-on:focusin.window="! $refs.panel.contains($event.target) && close()"
+				x-id="['dropdown-button']"
+			<?php endif; ?>
+		>
+			<?php inner_blocks(); ?>
```
Comment (bd-viget) [inline wp-content/themes/wp-starter/blocks/navigation-container/render.php:64]: Do we want to restrict the `allowedBlocks` in any way here? Just a thought, it's fine to leave as-is.

File: wp-content/themes/wp-starter/blocks/navigation-container/render.twig
Comment (bd-viget) [inline wp-content/themes/wp-starter/blocks/navigation-container/render.twig:1]: Same comments here as the PHP version.

**Review by bd-viget — APPROVED** [review]
Looks good! Just added a few notes, but I may have a few more we can go over during our meeting.

File: wp-content/themes/wp-starter/blocks/navigation-container/render.php (hunk 1 above)
Comment (nathan-schmidt-viget) [reply wp-content/themes/wp-starter/blocks/navigation-container/render.php:34]: Thanks for catching that!

**Pull request merged**